src/System.Memory/src/System/Buffers/Text/Utf8Constants.cs

```diff
@@ -32,5 +32,8 @@ internal static partial class Utf8Constants
 
         public const int DateTimeNumFractionDigits = 7;  // TimeSpan and DateTime formats allow exactly up to many digits for specifying the fraction after the seconds.
         public const int MaxDateTimeFraction = 9999999;  // ... and hence, the largest fraction expressible is this.
+
+        public const ulong BillionMaxUIntValue = 4294967295000000000; // uint.MaxValue * Billion, maximum value that can be split into two uint32 {1-10 digits}{9 digits}
```

Comment:
Nit: instead of hardcoding the result and needing a comment explaining it, did you consider just having:
```C#
public const ulong BillionMaxUIntValue = (ulong)uint.MaxValue * Billion; // maximum value that can be split into two uint32 {1-10 digits}{9 digits}
```